### DISCUSSION TITLE
Analysing the output of MODE across several dates

### DISCUSSION THREAD

**Discussion by ghost** [dtcenter/METplus#1523] [Statistical Computation] [answered]
Hi,

I'm looking to aggregate the output of MODE from multiple files to generate some seasonal statistics. It looks as though MET has the [MODE-Analysis Tool](https://met.readthedocs.io/en/latest/Users_Guide/mode-analysis.html), while I can't see it in METplus.

Is there any way to run the MODE-Analysis Tool in METplus or am I better off just writing my own script to summarize the results from multiple files of MODE output?

Alternatively is there any possible way for METplus save the output of MODE (from a season's worth of input data) in a single file rather than a file for each valid time?

Thanks,
Nick

**Comment by georgemccabe** [comment, marked as answer]
Hi Nick,

You are correct that there is no wrapper for MODEAnalysis in the latest METplus release (4.1.0). I created GitHub issue https://github.com/dtcenter/METplus/issues/1525 to implement it. It is likely that this wrapper will be made available in the next METplus release.

In the meantime, you are able to build a call to this tool using the UserScript wrapper. See [this link](https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/index.html#userscript) for examples of how to use the UserScript wrapper. You can set the USER_SCRIPT_COMMAND value to the command you want to build with any arguments. You can use filename template tags like {valid?fmt=%Y%m%d%H} to substitute the current run time values into the command:

`USER_SCRIPT_COMMAND = {MET_BIN_DIR}/mode_analysis -lookin {MODE_OUTPUT_DIR}/{valid?fmt=%Y%m%d%H}`

I don't believe there is any way to configure MODE to write all output for a season to the same file. This could be handled with a simple script that strips off the header line of each file except for the first file and concatenates the content into a single file. This could also be called using the UserScript wrapper so that it can be incorporated into a use case.

**Reply by georgemccabe** [reply]
Hi @nicholaslovedaybom, did this answer your question? If so, I'd like to mark the answer and lock the conversation to prevent future posts. If not, please let me know what else I can do to help.

**Reply by ghost** [reply]
Thanks @georgemccabe for the quick response. Yes that answers my question. That would be great if a MODEAnalysis wrapper could be added in the next METplus release.
